mesh/mesh.go

```diff
@@ -185,6 +186,77 @@ func (m *Mesh) SetLatestLayer(idx types.LayerID) {
 	}
 }
 
+func (m *Mesh) GetBlock(id types.BlockID) (*types.Block, error) {
+
+	blk, err := m.GetMiniBlock(id)
+	if err != nil {
+		m.Error("could not retrieve block %v from database %v", id, err)
+		return nil, err
+	}
+
+	return m.MiniBlockToBlock(blk)
+}
+
+func (m *Mesh) MiniBlockToBlock(blk *types.MiniBlock) (*types.Block, error) {
```

Comment:
At what point do you try to fetch the missing data (e.g. missing txs) from the node's neighbors? It seems that you only check in the db before you report a missing data in a block (for example in the flow that starts in the block listenen)

Reply:
all this stuff happens in the sync pkg this is just a utility method that fetches the block if it can and fails if not